tune_sklearn/tune_basesearch.py

```diff
@@ -296,7 +315,7 @@ def _fit(self, X, y=None, groups=None, **fit_params):
         classifier = is_classifier(self.estimator)
         cv = check_cv(cv=self.cv, y=y, classifier=classifier)
         self.n_splits = cv.get_n_splits(X, y, groups)
-        self.scoring = check_scoring(self.estimator, scoring=self.scoring)
+        _check_multimetric_scoring(self.estimator, self.scoring)
```

Comment:
There's a `_check_multimetric_scoring` in `__init__` -- do we need it here too?

Reply:
It's just because there's this test where they set the scoring after creating the object, and it won't error when they call fit with the wrong scoring. I'm not sure if we want to account for this case (where user sets scoring after making the object)

Reply:
I think we should make it consistent with scikit-learn, so whatever that entails I think should be the behavior we should try to get. I feel like we shouldn't need to change the tests to support this unless we want to intentionally change the design from scikit-learn

Reply:
Yeah it was intentional because like, we already need to check if the scoring is multimeric in the constructor so might as well call this method in the constructor too. I think this makes more sense but we could also just not call it until the user calls fit. @richardliaw what do you think?

Reply:
In tune-sklearn, can users set the scoring after creating the object?

Reply:
Yeah and you can also set it in sklearn too.

Reply:
This validates that behavior and should be fine to leave as is right?

Reply:
I think @anthonyhsyu is saying that we shouldn't error in the constructor if the scoring is not valid right?

Reply:
Ok, so I checked against scikit-learn and they perform all these checks in [`fit`](https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/model_selection/_search.py#L657), probably to handle both cases (if `scoring` is set when instantiating or set after that) with one call to `_check_multimetric_scoring`. Our `refit` metric is still set in `fit`, so would it be necessary in `__init__` then?

Reply:
Yeah we could just do the refit checks there too. I'll change it to that for now then

Reply:
Actually one of the tests fail now (`test_grid_search_groups`). In one of the loops, there's two consecutive calls to `fit()`, so we do `_check_multimetric_scoring` two times. The second time, because `self.scoring` has become a dict from the first call, it'll think that it's multimeric even if it actually isn't. I think it's cleaner to do checks in __init__, but this would mean changing the behavior.

Reply:
Maybe we could use this instead? 
https://github.com/scikit-learn/scikit-learn/blob/fd237278e/sklearn/model_selection/_search.py#L772
Basically pull it out of the dict if it's not multimetric